crates/wasmparser/src/readers/core/dylink0.rs

```diff
@@ -0,0 +1,106 @@
+use crate::{BinaryReader, Result, Subsection, Subsections};
+use std::ops::Range;
+
+/// Parser for the dynamic linking `dylink.0` custom section.
+pub type Dylink0SectionReader<'a> = Subsections<'a, Dylink0Subsection<'a>>;
+
+const WASM_DYLINK_MEM_INFO: u8 = 1;
+const WASM_DYLINK_NEEDED: u8 = 2;
+const WASM_DYLINK_EXPORT_INFO: u8 = 3;
+const WASM_DYLINK_IMPORT_INFO: u8 = 4;
+
+/// Represents a `WASM_DYLINK_MEM_INFO` field
+#[derive(Debug, Copy, Clone)]
+pub struct MemInfo {
+    /// Size of the memory area the loader should reserve for the module, which
+    /// will begin at `env.__memory_base`.
+    pub memory_size: u32,
+
+    /// The required alignment of the memory area, in bytes, encoded as a power
+    /// of 2.
+    pub memory_alignment: u32,
+
+    /// Size of the table area the loader should reserve for the module, which
+    /// will begin at `env.__table_base`.
+    pub table_size: u32,
+
+    /// The required alignment of the table area, in elements, encoded as a
+    /// power of 2.
+    pub table_alignment: u32,
+}
+
+#[allow(missing_docs)]
+#[derive(Debug)]
+pub struct ExportInfo<'a> {
+    pub name: &'a str,
+    pub flags: u32,
```

Comment:
Eventually, it would be cool to use `bitflags` to represent the possibilities described in https://github.com/WebAssembly/tool-conventions/blob/main/Linking.md.  Maybe add a TODO comment about that?

Reply:
Agreed!